website/static/js/formViewModel.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('osfHelpers');
+
+
+var ValidationError = function(errorHeader, errorMessage) {
+    this.errorHeader = errorHeader;
+    this.errorMessage = errorMessage;
+    this.errorLevel = 'warning';
+};
+
+var FormViewModel = function() {
+
+    // Abstract method each ViewModel must impliment. Expected behavior is to either return true
+    // or an array of ValidationError objects
+    self.isValid = function() {
+        throw "Not Implemented";
+    };
+
+    self.submit = function() {
+        try {
+            this.isValid();
+            return true; // Allow form to submit normally
+        } catch (ValidationErrors) {
```

Comment:
I see now that `ValidationErrors` is Array of messages thrown by `isValid`. However, this can still be improved by raising a `ValidationError` that can hold multiple messages. This will allow the error handling here to be more specific (only catch ValidationError) and subclasses throw an actual `Error` rather than an array.